### PR TITLE
Expose the ability to set a runtime's maximum stack size

### LINKED ISSUE #60: Error disposing a context after a stack overflow
I'm finding that I get an error if I try to dispose of a context after overflowing the stack. I've put together a minimal example [here](https://github.com/torywheelwright/quickjs-emscripten-repro-2). In the repro app, there are four test variants: sync/async, and using scope/not using scope variants. Neither of those factors seem to be significant, though. When I try to call QuickJSContext.dispose following the stack overflow exception, I get a trace like this:

```
$ yarn build && yarn start
yarn run v1.22.17
warning package.json: No license field
$ tsc -b .
Done in 0.19s.
yarn run v1.22.17
warning package.json: No license field
$ node dist
caught: RangeError: Maximum call stack size exceeded
    at wasm-function[23]:0x1900
    at wasm-function[237]:0xd46a
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
    at wasm-function[237]:0xca8c
Aborted(Assertion failed: list_empty(&rt->gc_obj_list), at: quickjs/quickjs.c,1983,JS_FreeRuntime)
/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/generated/emscripten-module.WASM_RELEASE_SYNC.js:16
                throw b;
                ^

RuntimeError: Aborted(Assertion failed: list_empty(&rt->gc_obj_list), at: quickjs/quickjs.c,1983,JS_FreeRuntime). Build with -s ASSERTIONS=1 for more info.
    at E (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/generated/emscripten-module.WASM_RELEASE_SYNC.js:143:71)
    at a (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/generated/emscripten-module.WASM_RELEASE_SYNC.js:226:47)
    at wasm-function[645]:0x2c1a9
    at wasm-function[981]:0x43e75
    at QuickJSFFI.a._QTS_FreeRuntime [as QTS_FreeRuntime] (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/generated/emscripten-module.WASM_RELEASE_SYNC.js:313:82)
    at Lifetime.disposer (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/module.js:185:22)
    at Lifetime.dispose (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/lifetime.js:71:18)
    at Scope.dispose (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/lifetime.js:220:26)
    at QuickJSRuntime.dispose (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/runtime.js:124:27)
    at Scope.dispose (/home/tory/quickjs-emscripten-repro-2/node_modules/quickjs-emscripten/dist/lifetime.js:220:26)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### PR DESCRIPTION
**Pull request by torywheelwright**
Break out quickjs JS_SetMaxStackSize as QuickJSRuntime.setMaxStackSize. Defines CONFIG_STACK_CHECK for quickjs, which enables stack size checks (otherwise JS_SetMaxStackSize has no effect).

Fixes #60

### CODE REVIEW

File: ts/module.ts [hunk 1]
```diff
@@ -296,6 +296,10 @@ export class QuickJSWASMModule {
       runtime.setInterruptHandler(options.interruptHandler)
     }
 
+    if (options.maxStackSize) {
+      runtime.setMaxStackSize(options.maxStackSize)
```
Comment (justjake) [inline ts/module.ts:300]: We also need a similar change in `module-asyncify.ts`. It's annoying to have a forked version of a function just for type shenanigans, maybe we should introduce a helper like `applyRuntimeOptions(runtime, options)`

**Review by justjake — APPROVED** [review]
Thanks for the help! Please see the fixup comment:

**Comment by torywheelwright** [comment]
I did the best I could with your suggestion without getting too deep into refactoring, and I also broke the maxStackSize option out on evalCode and evalCodeAsync.

**Comment by justjake** [comment]
Sorry it took so long to merge this. Life and whatnot 😅

**Pull request merged**